opentelemetry-sdk/src/growable_array.rs

```diff
@@ -87,6 +87,71 @@ impl<
         self.count + self.overflow.as_ref().map_or(0, Vec::len)
     }
 
+    /// Removes the element at a specific position (index) while preserving the order.
+    ///
+    /// This function performs the following operations:
+    ///
+    /// - If the index points to an element in the internal array (`inline`):
+    ///   - Removes the element at the specified index.
+    ///   - Shifts the remaining elements in the internal array to the left to fill the gap, preserving the order.
+    ///   - If an overflow vector exists:
+    ///     - Moves the first element from the overflow vector into the last position of the internal array, if available.
```

Comment:
Can we avoid doing this? It seems like an unnecessary expensive operation. We maintain the `count` of the items in the array. It should be okay to have the `count` less than the array length while also having the `vec`.